drivers/include/sdmmc/sdmmc.h

```diff
@@ -0,0 +1,1632 @@
+            uint32_t SD_SPEC:4;                 /**< SD Memory Card - Spec. Version [59:56] */
+            uint32_t SCR_STRUCTURE:4;           /**< SCR Structure [63:60] */
+        };
+        uint32_t value;
+    };
+} sdmmc_scr_t;
+
+#define SDMMC_SCR_ACMD_53_54_SUPPORT    (0b10000)   /**< @ref sdmmc_scr_t::CMD_SUPPORT
+                                                         Secure Receive/Send supported */
+#define SDMMC_SCR_ACMD_58_59_SUPPORT    (0b01000)   /**< @ref sdmmc_scr_t::CMD_SUPPORT
+                                                         Extension Register Multi-Block */
+#define SDMMC_SCR_ACMD_48_49_SUPPORT    (0b00100)   /**< @ref sdmmc_scr_t::CMD_SUPPORT
+                                                         Extension Register Single-Block */
+#define SDMMC_SCR_ACMD_23_SUPPORT       (0b00010)   /**< @ref sdmmc_scr_t::CMD_SUPPORT
+                                                         Set Block Count */
+#define SDMMC_SCR_ACMD_20_SUPPORT       (0b00001)   /**< @ref sdmmc_scr_t::CMD_SUPPORT
+                                                         Speed Class Control */
+
+/**
+ * @brief   Get Physical Layer Specification Version value from SCR value
+ *
+ * @param   scr   SCR register value of type sdmmc_scr_t
+ *
+ * | Value | Physical Layer Specification Version Number |
+ * |---|:---------------------------------|
+ * | 0 | Version 1.0 and 1.01 |
+ * | 1 | Version 1.10 |
+ * | 2 | Version 2.00 |
+ * | 3 | Version 3.0X |
+ * | 4 | Version 4.XX |
+ * | 5 | Version 5.XX |
+ * | 6 | Version 6.XX |
+ * | 7 | Version 7.XX |
+ * | 8 | Version 8.XX |
+ * | 9 | Version 9.XX |
+ */
+#define SDMMC_SCR_SD_SPEC(scr)  \
+    (scr.SD_SPEC + scr.SD_SPEC3 + (scr.SD_SPECX ? scr.SD_SPECX + 1 : scr.SD_SPEC4))
+
+/** @} */
+
+/**
+ * @brief   Data transfer types
+ */
+typedef enum {
+    SDMMC_MULTIBYTE, /**< Transfer 1 to READ_BL_LEN/WRITE_BL_LEN bytes */
+    SDMMC_BLOCK,     /**< Transfer block with size of READ_BL_LEN/WRITE_BL_LEN */
+    SDMMC_STREAM,    /**< Transfer until CMD12 is sent (MMC and 1-bit bus only) */
+} sdmmc_xfer_type_t;
+
+/**
+ * @brief   Transfer descriptor
+ *
+ * The transfer descriptor hold all information about a transfer such as the
+ * block size and the number of block.
+ */
+typedef struct {
+    sdmmc_xfer_type_t type;     /**< Type of the transfer */
+    uint8_t cmd_idx;            /**< Command index used for the transfer */
+    uint32_t arg;               /**< Command argument used for the transfer */
+    uint8_t resp_type;          /**< Type of expected response for the transfer */
+    bool auto_stop:1;           /**< Indicate that Auto CMD12/CMD23 can be used */
+    bool write:1;               /**< Indicate a write transfer */
+    uint16_t block_size;        /**< Size of a block or number of bytes for Byte transfer */
+    uint16_t block_num;         /**< Number of blocks to be transferred, 1 for Byte transfer */
```

Comment:
```suggestion
    uint32_t arg;               /**< Command argument used for the transfer */
    uint16_t block_size;        /**< Size of a block or number of bytes for Byte transfer */
    uint16_t block_num;         /**< Number of blocks to be transferred, 1 for Byte transfer */
    uint8_t cmd_idx;            /**< Command index used for the transfer */
    uint8_t resp_type;          /**< Type of expected response for the transfer */
    bool auto_stop:1;           /**< Indicate that Auto CMD12/CMD23 can be used */
    bool write:1;               /**< Indicate a write transfer */
```

sorting those by size removes the need for padding

Reply:
Hm, in this case it might be better to leave the members in the order of their context for documentation reasons. The `arg` and `resp_type` parameters refer to the specified `cmd_idx`.

Reply:
I have resorted a bit to 
```c
typedef struct {
    sdmmc_xfer_type_t type;     /**< Type of the transfer */
    bool write:1;               /**< Indicate a write transfer */
    sdmmc_cmd_t cmd_idx;        /**< Command index used for the transfer */
    sdmmc_resp_t resp_type;     /**< Type of expected response for the transfer */
    uint32_t arg;               /**< Command argument used for the transfer */
    uint16_t block_size;        /**< Size of a block or number of bytes for Byte transfer */
    uint16_t block_num;         /**< Number of blocks to be transferred, 1 for Byte transfer */
} sdmmc_xfer_desc_t;
```
which is a compromise between size and readability. It takes now 12 bytes, the smallest size possible with these members.

Reply:
Is the `bool auto_stop` no longer needed? If so, there should be no reason anymore for the bit field. Otherwise, I agree that using bit fields for `bool`s is indeed more readable than some `unsigned flags`.

Reply:
`auto_stop` is no longer used as a transmission type flag. It is now used implicitly if the peripheral driver supports it, which is indicated by the `periph_sdmmc_auto_cmd12` feature.